contribution_process.md

```diff
@@ -0,0 +1,149 @@
+
+1. It is implemented in the manner described in the issue.
+1. It meets the criteria specified in the contribution guide.
+
+If the criteria are not satisfied, the maintainers will comment on the issue
+specifically indicating what work is still required. If the contributor has
+questions, they should refer to this guide or ask the maintainers to clarify.
+Once the contributor makes the required changes, the maintainers will follow up.
+
+If the contributor is no longer actively engaged by responding to comments from
+maintainers or making code changes for more than 30 business days, the
+contribution is stale. If not, the pull request will be closed and the
+corresponding issue will be marked as ready for contribution.
+
+Contributions made without a GitHub issue are not prioritized to move to the
+next stage. Pull requests that are not linked to an issue and are more than six
+months old are considered stale and will be closed.
+
+## Ready for review stage
+
+Once the preliminary review criteria from the implementation stage have been
+met, the tracking GitHub issue is labeled as ready for review and added to a
+queue for maintainer review. The maintainers select issues for full review
+from this queue. The issues are prioritized for review publicly based on the
+following criteria in descending order of importance:
+
+1. Bugs, documentation changes, and feature requests are prioritized in that
+   order.
+1. Older contributions are selected before more recent ones.
+
+A maintainer must select the issue from the queue with the current highest
+priority. The prioritization of issues in the queue is reviewed on a regular
+cadence and is ultimately decided based on the maintainers' discretion.
+
+## Review stage
+
+Selecting a pull request for review moves the tracking GitHub issue to the
+under review stage. In this stage, the maintainers work with the
+contributor to perform a full review of the contribution with the goal of
+merging the pull request. Having a pull request selected for review does not
+guarantee how long it will be until it is merged. The maintainers will review no
+more than four pull requests at a time. The criteria for merging a pull request
+may include:
+
+1. The change is made in the right place and fits in the existing architecture.
+1. The change does not increase the maintainability in an unreasonable fashion.
+1. There is sufficient documentation, both in the code and for end users.
+1. The code is readable and follows best practices and conventions.
+
+If a pull request is not ready to merge, the maintainers will comment on the
+pull request with their feedback and questions for the contributor to address.
+If changes are requested, the contributor should make the changes within 15
+business days. If the contributor does not respond within 15 days, the pull
+request will be marked as stale and the contributor has 5 days to respond.
+If there is no response, the maintainers will complete the pull request.
```

Comment:
May complete? We should make sure we're clear on requirements once this enters the review stage. If there are substantial fixes required or a fundamental blocker is found with the proposed approach, we should try to time bound this.

Reply:
The idea is that the up front work before we open for community contribution and pre-review will give us enough confidence that we'll schedule and finish it.